flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -562,6 +567,7 @@ private void runMailboxLoop() throws Exception {
 
 	protected void afterInvoke() throws Exception {
 		LOG.debug("Finished task {}", getName());
+		getCompletionFuture().exceptionally(unused -> null).join();
```

Comment:
Two questions:

1. Should we not ignore the exception? Just in case of some unexpected exceptions/bugs/problems? I guess the expected exception is `InterruptedException`?
2. Shouldn't this be in a `while (true)` loop with ignoring interruptions?

Reply:
Good questions,

1. I think this future resulting exception shouldn't be reported here. It's already reported in `SourceStreamTask.processInput` using `mailboxProcessor.reportThrowable`
2. `CompletableFuture.join` is not interruptible, so shouldn't be a problem.

Reply:
1. 
> It's already reported in SourceStreamTask.processInput using mailboxProcessor.reportThrowable

I would rephrase it that "it SHOULD be reported in...". But what about handling our bugs, when it won't be reported? Could we make it more error prone? For example make sure that no exceptions should ever reach `StreamTask#getCompletionFuture().join()`? And if something would reach it here, it would be equivalent of an `IllegalStateException`?

Reply:
So you suggest to
1. move `.exceptionally(unused -> null)` from `StreamTask` to `SourceStreamTask` (or wrap in some other way)
2. in `StreamTask`, propagate failure as a programming error

Did I get you right?

If so, I think it introduces an unnecessary contract that `getCompletionFuture` never completes exceptionally and actually makes the code less error-prone (think of other `StreamTask` subclasses).

Reply:
ok, let's keep it as it is.